test/fail_compilation/fob1.d

```diff
@@ -0,0 +1,60 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/fob1.d(104): Error: variable `fob1.foo1.p` has undefined state and cannot be read
+---
+*/
+
+#line 100
+
+@live int* foo1()
+{
+    int* p = void;
+    return p;
+}
+
+/* TEST_OUTPUT:
+---
+fail_compilation/fob1.d(203): Error: variable `fob1.foo2.p` is left dangling at return
+---
+*/
+
+#line 200
+
+@live void foo2()
+{
+    int* p;
+    p = null;
```

Comment:
I'd also like to know. Pointing to null isn't dangling, pointing to stale memory is.

Reply:
`null` isn't recognized as special by the O/B system. It does not track `null` at all. This is because functions that may return `null` are not distinguishable from those that may not, rendering tracking `null` rather impotent.